src/Compilers/Core/Portable/DiagnosticAnalyzer/SymbolDeclaredCompilationEvent.cs

```diff
@@ -16,15 +17,23 @@ internal sealed class SymbolDeclaredCompilationEvent : CompilationEvent
     {
         private readonly Lazy<ImmutableArray<SyntaxReference>> _lazyCachedDeclaringReferences;
 
-        public SymbolDeclaredCompilationEvent(Compilation compilation, ISymbol symbol, SemanticModel? semanticModelWithCachedBoundNodes = null)
+        public SymbolDeclaredCompilationEvent(
+            Compilation compilation,
+            ISymbol symbol,
+            ISymbolInternal symbolInternal,
+            SemanticModel? semanticModelWithCachedBoundNodes = null)
             : base(compilation)
         {
             Symbol = symbol;
+            SymbolInternal = symbolInternal;
             SemanticModelWithCachedBoundNodes = semanticModelWithCachedBoundNodes;
             _lazyCachedDeclaringReferences = new Lazy<ImmutableArray<SyntaxReference>>(() => symbol.DeclaringSyntaxReferences);
         }
 
         public ISymbol Symbol { get; }
+
+        public ISymbolInternal SymbolInternal { get; }
```

Comment:
>public ISymbolInternal SymbolInternal { get; } [](http://example.com/codeflow?start=8&length=46)

I am not sure if we should be concerned about increasing the size of this object, but it looks like we always can get ISymbol from ISymbolInternal.